src/components/implementation/no_interface/vkernel/vk_api.c

```diff
@@ -1,71 +1,105 @@
 
-	printc("\tCreating and copying required initial capabilities\n");
 	/*
 	 * TODO: Multi-core support to create INITIAL Capabilities per core
 	 */
-	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITTHD_BASE, vkcinfo, vminfo->initthd);
+	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITTHD_BASE, vk_cinfo, initaep->thd);
+	assert(ret == 0);
+	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITHW_BASE, vk_cinfo, BOOT_CAPTBL_SELF_INITHW_BASE);
+	assert(ret == 0);
+
+	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITTCAP_BASE, vk_cinfo, initaep->tc);
 	assert(ret == 0);
-	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITHW_BASE, vkcinfo, BOOT_CAPTBL_SELF_INITHW_BASE);
+	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITRCV_BASE, vk_cinfo, initaep->rcv);
 	assert(ret == 0);
-	ret = cos_cap_cpy_at(vmcinfo, VM_CAPTBL_SELF_EXITTHD_BASE, vkcinfo, vminfo->exitthd);
+
+	ret = cos_cap_cpy_at(vmcinfo, VM_CAPTBL_SELF_SINV_BASE, vk_cinfo, vkinfo->sinv); 
 	assert(ret == 0);
+}
 
-	vminfo->inittcap = cos_tcap_alloc(vkcinfo);
-	assert(vminfo->inittcap);
+void
+vk_sl_thd_init(struct vms_info *vminfo)
+{
+	struct cos_compinfo *vk_cinfo = cos_compinfo_get(cos_defcompinfo_curr_get());
+	struct cos_defcompinfo *vmdci = &(vminfo->dci);
+	struct cos_compinfo *vmcinfo  = cos_compinfo_get(vmdci);
+	union sched_param spsameprio  = {.c = {.type = SCHEDP_PRIO, .value = (vminfo->id + 1)}};
+	union sched_param spsameC     = {.c = {.type = SCHEDP_BUDGET, .value = (VM_FIXED_BUDGET_MS * 1000)}}; 
+	union sched_param spsameT     = {.c = {.type = SCHEDP_WINDOW, .value = (VM_FIXED_PERIOD_MS * 1000)}};
+	int ret;
 
-	vminfo->initrcv = cos_arcv_alloc(vkcinfo, vminfo->initthd, vminfo->inittcap, vkcinfo->comp_cap, BOOT_CAPTBL_SELF_INITRCV_BASE);
-	assert(vminfo->initrcv);
+	vminfo->inithd = sl_thd_comp_init(vmdci, SL_THD_COMP_TCAP);
+	assert(vminfo->inithd);
 
-	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITTCAP_BASE, vkcinfo, vminfo->inittcap);
-	assert(ret == 0);
-	ret = cos_cap_cpy_at(vmcinfo, BOOT_CAPTBL_SELF_INITRCV_BASE, vkcinfo, vminfo->initrcv);
-	assert(ret == 0);
+	sl_thd_param_set(vminfo->inithd, spsameprio.v);
+	sl_thd_param_set(vminfo->inithd, spsameC.v);
+	sl_thd_param_set(vminfo->inithd, spsameT.v);
 
-	/*
-	 * Create send end-point in VKernel to each VM's INITRCV end-point
-	 */
-	vkinfo->vminitasnd[vminfo->id] = cos_asnd_alloc(vkcinfo, vminfo->initrcv, vkcinfo->captbl_cap);
-	assert(vkinfo->vminitasnd[vminfo->id]);
+	printc("\tsl_thd 0x%x created for thread = cap:%x, id=%u\n", (unsigned int)(vminfo->inithd),
+	       (unsigned int)sl_thd_thdcap(vminfo->inithd), (vminfo->inithd)->thdid);
 }
 
 void
 vk_iocaps_init(struct vms_info *vminfo, struct vms_info *dom0info, struct vkernel_info *vkinfo)
 {
-	struct cos_compinfo *vmcinfo = &vminfo->cinfo;
-	struct cos_compinfo *d0cinfo = &dom0info->cinfo;
-	struct cos_compinfo *vkcinfo = &vkinfo->cinfo;
+	struct cos_compinfo *vmcinfo = cos_compinfo_get(&vminfo->dci);
```

Comment:
I think it's fine because the library is mainly using `struct vms_info` as it's abstraction and acting on that data structure.